### ISSUE TITLE
Prompt user for confirmation regarding destructive actions in the context of staged changes

### ISSUE THREAD

**Issue by nullishamy** [jesseduffield/lazygit#3085]
**Is your feature request related to a problem? Please describe.**
When soft resetting (clearing the stage), lazygit will not prompt you to confirm, despite this being a destructive action.
When a stage is cleared, it is generally not possible to get those changes back (note: different from *unstaging*)

**Describe the solution you'd like**
Lazygit should prompt for all reset types, particularly ones where there are uncommitted changes. 
It can be argued that deleting commits is not destructive, since you can get them back through the reflog, which is fine.

**Describe alternatives you've considered**
Not deleting my stuff :(

**Additional context**
If it wasn't clear, this is the menu / items I am referring to: 
![image](https://github.com/jesseduffield/lazygit/assets/99221043/c6ef051e-5b0f-4f5a-b012-d3bfc1e9addf)

I would be open to PRing this in, if the maintainers believe it to be a worthwhile suggestion. I already went through and found where I
think [this issue arises](https://github.com/jesseduffield/lazygit/blob/8193731a828e1c966b5a850a08f24d07db6e2f9a/pkg/gui/controllers/workspace_reset_controller.go#L25).

**Comment by stefanhaller** [comment]
Of the three reset options, "hard" is the only one that is destructive (it discards your uncommitted changes), and I agree that it would be good to have a confirmation prompt for that one (but maybe only if there are any uncommitted changes?). The other two keep your uncommitted changes, so I feel that a confirmation is not needed for them. If you do want a prompt for these, please make it configurable, I would want to turn it off. I use soft reset pretty frequently, and I would find a prompt annoying.

Side note: it might be a good idea to swap the order of the "soft" and "mixed" commands, since "mixed" is the default for "git reset" on the command line, so it seems it should be what you get when you type `g`, `enter`.

**Comment by nullishamy** [comment]
Right. Yeah, I would agree it should only prompt if you have uncommitted changes present. I'm not *entirely* sure which option was selected that caused my changes to disappear (so, if you know how to restore changes lost by soft reset I would appreciate that), so some clarity about what causes what to happen would be good.

Also agree that it should be configurable, though I'd suggest opt-out as opposed to opt-in.

**Comment by stefanhaller** [comment]
As I said, "hard" is the only one that discards uncommitted changes (and there's no way to get them back as far as I know, so this is really the dangerous one). The other two don't touch your working copy, so I'm not sure what you mean by "changes lost by soft reset".

The only thing that you "lose" with soft or mixed reset is the commits, but as you said, it's easy to get them back via the reflog, and in lazygit it's especially easy: just press `z`.

**Comment by nullishamy** [comment]
Just tested quickly, and yeah soft reset doesn't touch the stage from what i can tell. So, with that, I agree that hard is the only one that *really* needs confirmation. Leaving this open for someone to PR, but I will try and get round to it if I have some spare time.
